### PR TITLE
chore(authenticator): Fix custom email test

### PR DESCRIPTION
**Pull request by dnys1**
*Issue #, if available:*

*Description of changes:*
- Fix custom email lambda test

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_authenticator/example/integration_test/pages/authenticator_page.dart [hunk 1]
```diff
@@ -81,22 +88,26 @@ abstract class AuthenticatorPage {
...
+    }
+    expect(inheritedBloc.authBloc.currentState, isA<Authenticated>());
   }
 
   /// Then I see User not found banner
-  Future<void> expectUserNotFound() async {
-    expectError('User does not exist.');
-  }
+  Future<void> expectUserNotFound() async => expectError(
+        Platform.isAndroid ? 'User not found' : 'User does not exist',
```
Comment (dnys1) [inline packages/amplify_authenticator/example/integration_test/pages/authenticator_page.dart:102]: Not sure why this discrepancy exists

File: packages/amplify_authenticator/example/integration_test/utils/mock_data.dart [hunk 2]
```diff
@@ -38,6 +38,6 @@ String generatePhone() => '+155501${randomPhoneDigits()}';
 
 String generateUsername() => 'flutter-user-${randomNumber()}';
 
-int randomNumber() => random.nextInt(500);
+int randomNumber() => random.nextInt(1 << 30);
```
Comment (Jordan-Nelson) [inline packages/amplify_authenticator/example/integration_test/utils/mock_data.dart:41]: 👍

**Review by Jordan-Nelson — APPROVED** [review]

File: packages/amplify_authenticator/example/integration_test/sign_in_with_email_test.dart [hunk 3]
```diff
@@ -85,62 +82,54 @@ void main() {
...
-      late String code = '';
-
-      Completer codeCompleter = Completer<ConfirmationCodeTestRun>();
 
       await loadAuthenticator(tester: tester, authenticator: authenticator);
       SignInPage signInPage = SignInPage(tester: tester);
       ConfirmSignUpPage confirmSignUpPage = ConfirmSignUpPage(tester: tester);
 
-      String subscriptionDocument = '''subscription MyMutation {
+      const subscriptionDocument = '''subscription {
```
[the remark this answers is not in the document]
Comment (dnys1) [reply packages/amplify_authenticator/example/integration_test/sign_in_with_email_test.dart:90]: Nice, I'll update it. I think I was working on an outdated branch.

File: packages/amplify_authenticator/example/integration_test/pages/authenticator_page.dart (hunk 1 above)
Comment (Jordan-Nelson) [reply packages/amplify_authenticator/example/integration_test/pages/authenticator_page.dart:102]: Ran into this on another test. Amplify-Android defines the messages in the library. I believe iOS is using the message from Cognito.

**Review by Jordan-Nelson — APPROVED** [review]

File: packages/amplify_authenticator/example/integration_test/sign_in_mfa_test.dart [hunk 4]
```diff
@@ -108,7 +108,7 @@ void main() {
       await confirmSignInPage.navigateToSignIn();
 
       // Then I see "Sign in"
-      await signInPage.expectSignIn();
+      signInPage.expectScreen(AuthScreen.signin);
```
[the remark this answers is not in the document]
Comment (haverchuck) [reply packages/amplify_authenticator/example/integration_test/sign_in_mfa_test.dart:111]: 👍

File: packages/amplify_authenticator/example/integration_test/utils/data_utils.dart [hunk 5]
```diff
@@ -103,61 +104,35 @@ Future<AdminCreateUserResponse?> adminCreateUser(
...
+        return ConfirmSignUpResponse.fromJson(json.cast());
+      })
+      .where((event) => event.username == username)
+      .map((event) => event.currentCode)
+      // When multiple Cognito events happen in a test, we must use the newest
+      // code, since the others will have been invalidated.
+      //
+      // Instead of taking a fixed number here, we take the most recent code
+      // after 5 seconds, which is hopefully a more general solution.
+      .audit(const Duration(seconds: 5))
```
Comment (haverchuck) [inline packages/amplify_authenticator/example/integration_test/utils/data_utils.dart:136]: I don't think I've used stream_transform - looks useful. Thanks!

**Review by haverchuck — APPROVED** [review]

**Pull request merged**